src/group_theory/schreier.lean

```diff
@@ -144,4 +142,35 @@ begin
   ... = H.index * group.rank G : congr_arg ((*) H.index) hS₀,
 end
 
+variables (G)
+
+/-- If `G` has `n` commutators `[g₁, g₂]`, then `|G'| ∣ [G : Z(G)] ^ ([G : Z(G)] * n + 1)`,
+where `G'` denotes the commutator of `G`. -/
+lemma card_commutator_dvd_index_center_pow [finite {g | ∃ g₁ g₂ : G, ⁅g₁, g₂⁆ = g}] :
+  nat.card (commutator G) ∣
+    (center G).index ^ ((center G).index * nat.card {g | ∃ g₁ g₂ : G, ⁅g₁, g₂⁆ = g} + 1) :=
+begin
+  -- First handle the case when `Z(G)` has infinite index and `[G : Z(G)]` is defined to be `0`
+  by_cases hG : (center G).index = 0,
+  { simp_rw [hG, zero_mul, zero_add, pow_one, dvd_zero] },
+  -- Rewrite as `|Z(G) ∩ G'| * [G' : Z(G) ∩ G'] ∣ [G : Z(G)] ^ ([G : Z(G)] * n) * [G : Z(G)]`
+  rw [←((center G).subgroup_of (commutator G)).card_mul_index, pow_succ'],
+  -- We have `h1 : [G' : Z(G) ∩ G'] ∣ [G : Z(G)]`
+  have h1 := relindex_dvd_index_of_normal (center G) (commutator G),
+  -- So we can reduce to proving `|Z(G) ∩ G'| ∣ [G : Z(G)] ^ ([G : Z(G)] * n)`
+  refine mul_dvd_mul _ h1,
+  -- We have `h2 : rank (Z(G) ∩ G') ≤ [G' : Z(G) ∩ G'] * rank G'` by Schreier's lemma
+  have h2 := rank_le_index_mul_rank (ne_zero_of_dvd_ne_zero hG h1),
+  -- We have `h3 : [G' : Z(G) ∩ G'] * rank G' ≤ [G : Z(G)] * n` by `h1` and `rank G' ≤ n`
+  have h3 := nat.mul_le_mul (nat.le_of_dvd (nat.pos_of_ne_zero hG) h1) (rank_commutator_le_card G),
+  -- So we can reduce to proving `|Z(G) ∩ G'| ∣ [G : Z(G)] ^ rank (Z(G) ∩ G')`
+  refine dvd_trans _ (pow_dvd_pow (center G).index (h2.trans h3)),
+  -- `Z(G) ∩ G'` is abelian, so it enough to prove that `g ^ [G : Z(G)] = 1` for `g ∈ Z(G) ∩ G'`
+  apply card_dvd_exponent_pow_rank' _ (λ g, _),
+  -- `Z(G)` is abelian, so `g ∈ Z(G) ∩ G' ≤ G' ≤ ker (transfer : G → Z(G))`
+  have := abelianization.commutator_subset_ker (monoid_hom.transfer_center_pow' hG) g.1.2,
```

Comment:
Do we have [monoid_hom.transfer_center_pow](https://leanprover-community.github.io/mathlib_docs/group_theory/transfer.html#monoid_hom.transfer_center_pow) for general normal subgroups instead of just the center?

Reply:
The fact that the transfer homomorphism `G -> Z(G)` is given by the `[G : Z(G)]`th power map is pretty special to `Z(G)`.

Reply:
You only have a transfer homomorphism `G -> H` if `H` is abelian. And it is quite special for this transfer homomorphism to be given by the `[G : H]`th power map. The most general condition I know is [`transfer_eq_pow`](https://leanprover-community.github.io/mathlib_docs/group_theory/transfer.html#monoid_hom.transfer_eq_pow).

Reply:
Thanks! It's indeed rather special. Normality only guarantees the image of the power map lies in the subgroup, but it's not necessarily a homomorphism, even when the subgroup is both normal and abelian (just consider a dihedral group and the abelian normal subgroup of index 2 consisting of rotations, where the square of every reflection is trivial but the square of a rotation can be nontrivial, yet the square of the product of a reflection and a rotation (which is a reflection), is always trivial). Sorry for the noise!

Reply:
Ah, that's why you mentioned normality :)